qiskit_experiments/framework/analysis_result_table.py

```diff
@@ -204,16 +204,26 @@ def clear(self):
         with self._lock:
             self._data = pd.DataFrame(columns=self.DEFAULT_COLUMNS)
 
-    def copy(self):
+    def copy(self, new_ids: bool = True):
```

Comment:
Nothing uses `new_ids=False`, so should we just not include that option for now? It's a question of why a user is calling `copy`. If making a copy, it seems likely the user would want to change something and then in that case likely that new IDs should be used?